### PR TITLE
Added secure characteristic test.

### PR DESCRIPTION
**Pull request by cmumford**
This is a test to connect to a secure device and read/write characteristics.
One service (age) is encrypted, and heart-rate is unencrypted.

### CODE REVIEW

File: characteristic_secure/device_code.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+  NRF.on('connect', (addr) => {
+    debug = `onconnect: ${addr}`;
+    isConnected = true;
+    updateUI();
+  });
+
+  NRF.setSecurity({
+    display: deviceHasDisplay(),  // Can this device display a passkey?
+    keyboard: false,              // Can this device enter a passkey?
```
Comment (reillyeon) [inline characteristic_secure/device_code.js:134]: Use `deviceHasKeyboard()`.

File: characteristic_secure/web_app.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+async function startTest() {
+  clearStatus();
+  logInfo('Starting test');
+
+  $('btn_start_test').disabled = true;
+  $('btn_load_code').disabled = true;
+
+  try {
+
```
Comment (reillyeon) [inline characteristic_secure/web_app.js]: Extra newline.

File: characteristic_secure/web_app.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+      dataView = await characteristic.readValue();
+      val = dataView.getUint8(0, /*littleEndian=*/true);
+      assertEquals(110, val, 'incorrect maximum heart rate');
+      logInfo('readback success.');
+    }
+
+    const options = {
+      filters: [{ services: [nordicUARTService] }],
+      optionalServices: [ageServiceNumber,
+        BluetoothUUID.getService(heartRateServiceName)]
```
Comment (reillyeon) [inline characteristic_secure/web_app.js]: I appreciate the desire to hit 100% function coverage in this test suite but these functions are already well tested by the fully automated Web Platform Tests and using them here adds confusion to an already complex test case.

**Review by reillyeon — CHANGES_REQUESTED** [review]

File: characteristic_secure/web_app.js (hunk 2 above)
Comment (cmumford) [reply characteristic_secure/web_app.js]: Missed this. I'll get it before landing.

File: characteristic_secure/device_code.js (hunk 1 above)
Comment (cmumford) [reply characteristic_secure/device_code.js:134]: I deleted the function as it didn't improve clarity.

File: characteristic_secure/web_app.js (hunk 3 above)
Comment (cmumford) [reply characteristic_secure/web_app.js]: Done.

**Review by reillyeon — APPROVED** [review]

**Pull request merged**